### PR TITLE
Run Forseti in Docker on Kubernetes or Container-Optimized OS (Proof of Concept)

### PR DESCRIPTION
**Pull request by angelsungoogle**
Many thanks to @Red-Five for pulling up these changes and fixes! Please feel free to contribute.

@andreyk-code is Kubernetes subject matter expert and will the peer reviewer for @Red-Five work here.

Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](https://forsetisecurity.org/docs/latest/develop/dev/testing.html) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blueandgold** [comment]
@angelsungoogle Can you please make sure that you have sign the CLA and that you have configured your github env to use the correct email?  I will follow-up with you offline about this, in more detail.

**Comment by Red-Five** [comment]
> 😕 The bad news is that it appears that one or more commits were authored or co-authored by someone other than the pull request submitter. We need to confirm that all authors are ok with their commits being contributed to this project. Please have them confirm that here in the pull request.

Confirming that i would like my changes initial changes merged into the docker-poc branch. 

Kubernetes (GKE) files will be added shortly.

**Comment by Red-Five** [comment]
@angelsungoogle  This pull request is to dev. I thought we were going to keep a separate docker-poc branch for now? I'm OK either way, however you think best.

**Comment by angelsungoogle** [comment]
@Red-Five we do have a separate branch docker-poc, which is the branch this PR is based on. For collaboration, people could continue to push to the docker-poc branch. I'm just keeping a PR here, since eventually we will be adding these to dev, when we are ready.

**Comment by angelsungoogle** [comment]
[the docker poc branch](https://github.com/GoogleCloudPlatform/forseti-security/tree/docker-poc)

**Comment by Red-Five** [comment]
Forseti Server Tasks
- [x] cloudbuild.yaml file path fix
- [x] cloudbuild.yaml unittest step
- [x] base dockerfile Google SDK Install
- [x] Add docker_entrypoint.sh
- [ ] docker_entrypoint.sh: add var to toggle debugging
- [ ] docker_entrypoint.sh: add var to control which services to run in container
- [ ] Add Kubernetes manifest file
- [ ] Cron job on Container Optimized OS (cos)
- [ ] Cron job on GKE

Forseti Client Tasks
- [ ] Define deployment architecture for cos and GKE (sidecar? separate pod?)
- [ ] Verify client server connectivity

**Comment by angelsungoogle** [comment]
manually verified CLA, verified that changes committed are agreed by the author via GVC

**Comment by angelsungoogle** [comment]
Please note that the current Kubernetes PoC is hosts Forseti intermittently: when the scheduled cron job starts, a pod is created to host Forseti and run the pre-defined cron job script; at the end of the cron job this pod is deleted.

**Comment by Red-Five** [comment]
The files are still a little rough around the edges. 
Needs comment clean up and 
either fix the cos startup script to include the new cloud sql env vars or remove the cos approach altogether and just focus on k8s.

**Comment by Red-Five** [comment]
> The next steps will likely be to make a forseti.service.template.yaml following the [GKE Service Tutorial](https://cloud.google.com/kubernetes-engine/docs/concepts/service), and add something like a if statement [here](https://github.com/GoogleCloudPlatform/forseti-security/pull/2335/files#diff-b5f19dfbf5be42d0523fe23abee37d4aR58)

Very rough draft, not at ready yet: https://github.com/GoogleCloudPlatform/forseti-security/pull/2440

**Comment by angelsungoogle** [comment]
Thanks Duncan!
@blueandgold our favorite client has just done the work for us, right before Christmas XD

**Comment by Red-Five** [comment]
This might be work looking into. https://cloud.google.com/knative/

**Comment by angelsungoogle** [comment]
@blueandgold

**Comment by Red-Five** [comment]
https://github.com/GoogleCloudPlatform/forseti-security/pull/2590

File: install/docker/base [hunk 1]
```diff
@@ -24,5 +24,6 @@ RUN apt-get update -qq 1> /dev/null && \
     apt-get install -qq -y $(cat /tmp/apt_packages.txt | grep -v "#" | xargs) 1> /dev/null && \
     rm -rf /var/lib/apt/lists/*
 
-# Install the CloudSDK for `gcloud`.
-RUN curl -sSL https://sdk.cloud.google.com 1> /dev/null | bash
+# Install Google Cloud SDK
+RUN curl -sSL https://sdk.cloud.google.com | bash
```
[the remark this answers is not in the document]
Comment (blueandgold) [reply install/docker/base]: Run `chmod` or `chown`?

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as all commit authors have contributed to this project.

**Pull request merged**